scipy/integrate/_cubature.py

```diff
@@ -0,0 +1,617 @@
+import math
+import heapq
+import itertools
+import functools
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+import numpy as np
+
+from numpy.polynomial.polynomial import Polynomial
+
+
+def cub(f, a, b, rule, rtol=1e-05, atol=1e-08, limit=10000, args=()):
```

Comment:
args is to allow `f(x, *args)`? Then let's go the whole way and allow `f(x, *args, **kwds)`. (remember to not use `{}` as the default).